test/unit/testExecutorCurrentNameReason.js

```diff
@@ -0,0 +1,241 @@
+                currentNameReason: 'Marriage'
+            };
+            testErrors = [];
+        });
+
+        it('returns the correct data and errors', (done) => {
+            const ExecutorCurrentNameReason = steps.ExecutorCurrentNameReason;
+            const [ctx, errors] = ExecutorCurrentNameReason.handlePost(testCtx, testErrors);
+
+            expect(ctx.list[0]).to.deep.equal({
+                currentNameReason: 'Marriage',
+                isApplying: true,
+            });
+            expect(errors).to.deep.equal(testErrors);
+            done();
+        });
+
+        it('returns the correct data for option:"other"', (done) => {
+            testCtx.currentNameReason = 'other';
+            testCtx.otherReason = 'it was a dare';
+            const ExecutorCurrentNameReason = steps.ExecutorCurrentNameReason;
+            const [ctx, errors] = ExecutorCurrentNameReason.handlePost(testCtx, testErrors);
+
+            expect(ctx.list[0]).to.deep.equal({
+                currentNameReason: 'other',
+                isApplying: true,
+                otherReason: 'it was a dare'
+            });
+            expect(errors).to.deep.equal(testErrors);
+            done();
+        });
+    });
+
+    describe('recalcIndex()', () => {
+        it('returns the index when an executor can be found', (done) => {
+            const testCtx = {
+                list: [
+                    {isApplying: true, hasOtherName: true},
+                    {isApplying: true, hasOtherName: false},
+                    {isApplying: true, hasOtherName: true}
+                ]
+            };
+            const ExecutorCurrentNameReason = steps.ExecutorCurrentNameReason;
+            const index = ExecutorCurrentNameReason.recalcIndex(testCtx, 0);
+
+            expect(index).to.equal(2);
+            done();
+        });
+
+        it('returns -1 when an executor cannot be found', (done) => {
+            const testCtx = {
+                list: []
+            };
+            const ExecutorCurrentNameReason = steps.ExecutorCurrentNameReason;
+            const index = ExecutorCurrentNameReason.recalcIndex(testCtx, 0);
+
+            expect(index).to.equal(-1);
+            done();
+        });
+    });
+
+    describe('nextStepOptions()', () => {
+        it('returns the next step options', (done) => {
+            const testCtx = {
+                index: 1,
+                executorsWrapper: new ExecutorsWrapper()
+            };
+            const ExecutorCurrentNameReason = steps.ExecutorCurrentNameReason;
+            const nextStepOptions = ExecutorCurrentNameReason.nextStepOptions(testCtx);
+
+            expect(nextStepOptions).to.deep.equal({
+                options: [
+                    {key: 'continue', value: true, choice: 'continue'},
+                ],
+            });
+            done();
+        });
+    });
+
+    describe('action()', () => {
+        it('removes the correct values from the context', (done) => {
```

Comment:
This test should also check that `ctx.index` has been removed from the returned object.